src/psyclone/tests/psyir/nodes/assignment_test.py

```diff
@@ -158,3 +160,46 @@ def test_is_array_range():
     array_ref = ArrayReference.create(symbol, [my_range, int_one.copy()])
     assignment = Assignment.create(array_ref, one.copy())
     assert assignment.is_array_range
+
+    # Check when lhs consists of various forms of structure access
+    grid_type = StructureType.create([
```

Comment:
Could you please add an example that is a structure, but does not have an array range.

Reply:
Done.

Reply:
Great.